pkg/prometheus/resource_selector.go

```diff
@@ -691,6 +691,15 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			continue
 		}
 
+		for _, rl := range sc.Spec.MetricRelabelConfigs {
+			if rl.Action != "" {
+				if err = validateRelabelConfig(rs.p, *rl); err != nil {
+					rejectFn(sc, err)
+					continue
+				}
+			}
+		}
```

Comment:
```suggestion
		if err = validateRelabelConfigs(rs.p, sc.Spec.MetricRelabelConfigs); err != nil {
			rejectFn(sc, fmt.Errorf("metricRelabelConfigs: %w", err))
			continue
		}
```

Reply:
Rebased and code updated to use the _validateRelabelConfigs_ function